AnnotatoBackend/Sources/App/WebSocketControllers/OfflineToOnlineWebSocketController.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+import Vapor
+import AnnotatoSharedLibrary
+import FluentKit
+
+class OfflineToOnlineWebSocketController {
+    private static var logger = Logger(label: "OfflineToOnlineWebSocketController")
+
+    static func handleOfflineToOnlineResolution(userId: String, data: Data, db: Database) async {
+        do {
+            Self.logger.info("Processing offline to online data...")
+
+            let message = try JSONCustomDecoder().decode(AnnotatoOfflineToOnlineMessage.self, from: data)
+
+            switch message.mergeStrategy {
+            case .keepServerVersion:
+                await handleKeepServerVersion(userId: userId, db: db, message: message)
+            case .overrideServerVersion:
+                await Self.handleOverrideServerVersion(userId: userId, db: db, message: message)
+            }
+        } catch {
+            Self.logger.error("Error when handling incoming offline to online data. \(error.localizedDescription)")
+        }
+    }
+
+    private static func handleKeepServerVersion(
+        userId: String,
+        db: Database,
+        message: AnnotatoOfflineToOnlineMessage
+    ) async {
+        do {
+            let responseDocuments = try await DocumentsDataAccess
```

Comment:
## Test

Step 1: POST to `localhost:8080/documents`

```
{
    "id": "aa763c89-0b48-4a3a-9094-7ab1ad25c7e6",
    "name": "test document",
    "ownerId": "4YDoKKGWLuNWvLZixnJ2TQG2ZAF2",
    "baseFileUrl": "path/to/doc",
    "annotations": []
}
``` 

Step 2: Connected to ws://localhost:8080/ws/4YDoKKGWLuNWvLZixnJ2TQG2ZAF2

```
{
    "type": "offlineToOnline",
    "mergeStrategy": "keepServerVersion",
    "lastOnlineAt": "2022-04-02T18:15:45Z",
    "documents": [
        {
            // "name": "CS3217-lec1a-Introduction to CS3217.pdf",
            "name": "oink",
            "ownerId": "user",
            "baseFileUrl": "url",
            "id": "aa763c89-0b48-4a3a-9094-7ab1ad25c7e6",
            "annotations": []
            // "deletedAt": "2022-04-02T18:15:45Z"
        }
    ],
    "annotations": []
}
```

Response: 

```
{"mergeStrategy":"keepServerVersion","lastOnlineAt":"2022-04-02T18:15:45Z","documents":[],"annotations":[],"type":"offlineToOnline"}
```

Behavior: 

Does not update the document in the server which is correct.

However, the response arrays are empty. Shouldn't it contain the server's versions so that the sender can update their local versions accordingly? We would want to document name and baseFIleUrl to be the same as the server's versions.

I checked the code and its the same for annotations.

Reply:
For keep server, for documents

- It's not possible to get access to new server documents while you're offline
  - Even if other people post new documents to the server, there's no way for you to import while offline (aka no new shared documents)
  - There's no way for you to post new documents to the server (aka no new own documents)

Reply:
I think you misunderstood. The one you are talking about above is the one I asked below in the next comment. 

But for this one, the server's version of the document with id `aa763c89-0b48-4a3a-9094-7ab1ad25c7e6` has the name `test document`.  This was created using the `POST` request. 

Then the user sends an `offline to online` message updating the document name to `oink`.  This means locally the user's document with id `aa763c89-0b48-4a3a-9094-7ab1ad25c7e6` has the name `oink`.

After the server does whatever it needs to do, does it need to send back the document with the name `test document` as it is the one we have to update in the user's local database? Or else the document with id  `aa763c89-0b48-4a3a-9094-7ab1ad25c7e6` will continue to have the name `oink`.